hvac/api/auth_methods/approle.py

```diff
@@ -286,7 +292,7 @@ def generate_secret_id(
             mount_point=mount_point,
             role_name=role_name,
         )
-        return self._adapter.post(url=api_path, json=params)
+        return self._adapter.post(url=api_path, json=params, wrap_ttl=wrap_ttl)
```

Comment:
should `wrap_ttl` be part of the payload? I don't see a `wrap_ttl` parameter on [requests](https://github.com/python/typeshed/blob/b0dc6a340523638df63e1790741689fda693a297/stubs/requests/requests/sessions.pyi#L215-L233)

Reply:
it shouldn't be in the payload in this case. `_adapter` is our adapter rather an adapter from `requests`. The `post` method will call the `request` method:
https://github.com/hvac/hvac/blob/develop/hvac/adapters.py#L112-L123

Our `request` method looks for `wrap_ttl` and if it's supplied, it pops it out of the kwargs and makes it into a header:
https://github.com/hvac/hvac/blob/develop/hvac/adapters.py#L288-L290

Reply:
I also added both integration and unit test coverage for `wrap_ttl`, if you want to see it in action